gym_anytrading/envs/trading_env.py

```diff
@@ -49,14 +51,23 @@ def __init__(self, df, window_size):
         self._first_rendering = None
         self.history = None
 
+        # create cache for faster training
+        self._observation_cache = []
```

Comment:
Does this cache really improve the speed? It only stores multiple slices of self.signal_features. I haven't tested the speed but I don't think it is slow to read a slice of self.signal_features in each step. Also, self.signal_features is calculated once in the class constructor and somehow is already cached.

Reply:
not necessary for 2D shape.

for 1D you have to run the function ".flatten()", this reduced the performance a bit. (maybe only 1 to 2 %)